File: 프로그래머스/Level 0/분수의 덧셈-solve.java
class Solution {
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int resNumer = numer1 * denom2 + numer2 * denom1;
        int resDenom = denom1 * denom2;
        
        for (int div = Math.min(resNumer, resDenom); div >= 2; div--) {
            if (resNumer % div == 0 && resDenom % div == 0) {
                resNumer /= div;
                resDenom /= div;
            }
        }
        
        return new int[] {resNumer, resDenom};
    }
}

File: 프로그래머스/Level 0/분수의 덧셈-solve.py
def solution(numer1, denom1, numer2, denom2):
    res_numer = numer1 * denom2 + numer2 * denom1
    res_denom = denom1 * denom2 
    
    for div in range(min(res_numer, res_denom), 1, -1): 
        if res_numer % div == 0 and res_denom % div == 0:
            res_numer //= div
            res_denom //= div 
    
    return [res_numer, res_denom]